src/v/cloud_io/basic_cache_service_api.h

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2021 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "base/seastarx.h"
+#include "base/units.h"
+
+#include <seastar/core/file.hh>
+#include <seastar/core/future.hh>
+#include <seastar/core/io_priority_class.hh>
+#include <seastar/core/iostream.hh>
+#include <seastar/core/lowres_clock.hh>
+
+#include <filesystem>
+#include <iterator>
+#include <optional>
+#include <set>
+#include <string_view>
```

Comment:
nit: iterator, set and string_view appear unused.

Reply:
https://github.com/redpanda-data/redpanda/pull/23810